src/main/java/com/yahoo/bullet/tracing/FilterRule.java

```diff
@@ -36,7 +36,8 @@ public FilterRule(String input, Map configuration) throws ParsingException {
 
     @Override
     public boolean consume(BulletRecord record) {
-        if (!specification.isAcceptingData() || !specification.filter(record)) {
+        // If rule is expired, not accepting data or does not match filters, don't consume...
+        if (isExpired() || !specification.isAcceptingData() || !specification.filter(record)) {
```

Comment:
Thought about adding that but since a query generally runs longer than a tick, the number of the times we would check this boolean (and do the time check) while the query is running would be >> the number of times we *don't* do the time check after the query has expired and before the tick.